src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java

```diff
@@ -260,10 +261,46 @@ protected static String buildAndConditions(Map<String, String> andConditions) {
     //Hilsfunktion zum setzen der Parameter
     protected static void registerAndConditions(Map<String, String> andConditions, NativeQuery query) {
         for (Map.Entry<String, String> andCondition : andConditions.entrySet()) {
-            query.setParameter(andCondition.getKey(), andCondition.getValue());
+
+            if (andCondition.getKey().equals("QuelleBisJahrhundert") || andCondition.getKey().equals("QuelleVonJahrhundert") || andCondition.getKey().equals("VonJahrhundert") || andCondition.getKey().equals("BisJahrhundert") ) {
```

Comment:
Why is this necessary? a generic function like this should not have special hardcoded handling for field names. If it is necessary to handle certain columns in a special way, this should be passed to the function as parameter as well.